builder/repo_control.go

```diff
@@ -224,6 +224,34 @@ func (b *Builder) SetExcludesRepo(reponame, pkgs string) error {
 	return DNFConf.SaveTo(b.Config.Builder.DNFConf)
 }
 
+// OverrideDnfConf creates a temporary dnf conf file based on the original
+// and overrides repo/URL settings.
+func (b *Builder) OverrideDnfConf(repoURLs map[string]string) (*os.File, error) {
+	// Create tmp dnf config based on original
+	tmpConf, err := ioutil.TempFile(os.TempDir(), "mixerConf")
+	if err != nil {
+		return tmpConf, err
+	}
+	origConf, err := ioutil.ReadFile(b.Config.Builder.DNFConf)
+	if err != nil {
+		return tmpConf, err
+	}
+	err = ioutil.WriteFile(tmpConf.Name(), origConf, 0644)
```

Comment:
You already have a file descriptor open for writing, you don't need to open it again here.